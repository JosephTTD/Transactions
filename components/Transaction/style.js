import styled from "styled-components";

const Container = styled.div`
  background: transparent;
  border-radius: 5px;
  color: black;
  padding: 5px 0px;
  display: flex;
  justify-content: space-between;
  align-items: center;

  p {
    font-size: 14px;
  }

  sup {
    font-size: 10px;
    color: grey;
  }
`;

const Content = styled.div`
  display: flex;
  flex-direction: row;
  font-weight: 300;
  align-items: center;

  span {
    background: black;
    color: ${(props) => props.crest};
    height: 30px;
    width: 30px;
    display: flex;
    justify-content: center;
    align-items: center;
    border-radius: 100px;
    margin-right: 15px;
  }

  p {
    margin: 0;
    &:first-child {
      color: grey;
      font-size: 10px;
    }

    &:nth-child(2) {
      font-size: 15px;
    }
  }
`;

export { Container, Content };
