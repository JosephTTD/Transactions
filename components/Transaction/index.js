import React from "react";
import { Container, Content } from "./style";

function Transaction({ name, value, currency, category, crest }) {
  var randomColor = Math.floor(Math.random() * 16777215).toString(16);
  return (
    <Container>
      <Content crest={crest}>
        {name && <span>{name[0]}</span>}

        <div>
          {category && <p>{category}</p>}
          {name && <p>{name}</p>}
        </div>
      </Content>

      {value &&  (
        <p>
          {value}
          <sup> {currency}</sup>
        </p>
      )}
    </Container>
  );
}

export default Transaction;
