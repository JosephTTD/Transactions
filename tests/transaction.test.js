import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Transaction from "../components/Transaction";

const component = shallow(
  <Transaction
    name="Kidz Bop"
    value="123"
    currency="USD"
    category="Entertainment"
    crest="#fe2651"
  />
);

describe("Transaction", () => {
  it("renders correctly", () => {
    expect(toJson(component)).toMatchSnapshot();
  });

  it("should render the prop values correctly", () => {
    expect(component.find("span").text()).toEqual("K");
    expect(component.find("p").at(0).text()).toEqual("Entertainment");
    expect(component.find("p").at(1).text()).toEqual("Kidz Bop");
    expect(component.find("p").at(2).text()).toEqual("123 USD");
  });
});
