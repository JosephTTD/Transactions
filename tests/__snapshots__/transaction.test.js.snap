// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transaction renders correctly 1`] = `
<styled.div>
  <styled.div
    crest="#fe2651"
  >
    <span>
      K
    </span>
    <div>
      <p>
        Entertainment
      </p>
      <p>
        Kidz Bop
      </p>
    </div>
  </styled.div>
  <p>
    123
    <sup>
       
      USD
    </sup>
  </p>
</styled.div>
`;
